frameworks/mlr3automl/exec.R

```diff
@@ -0,0 +1,58 @@
+library(mlr3)
+library(mlr3learners)
+library(mlr3extralearners)
+library(mlr3tuning)
+library(mlr3pipelines)
+library(mlr3automl)
+library(mlr3oml)
+
+run <- function(train_file, test_file, target.index, type, output_predictions_file, cores, time.budget, seed, name) {
+  start_time = Sys.time()
+  set.seed(seed)
+  
+  train <- mlr3oml::read_arff(train_file)
+  colnames(train) <- make.names(colnames(train))
+  target <- colnames(train)[target.index]
+
+  test <- mlr3oml::read_arff(test_file)
+  colnames(test) <- make.names(colnames(test))
+  
+  print(paste("Finished loading data after ", Sys.time() - start_time, " seconds"))
+  remaining_budget = as.integer(start_time - Sys.time() + time.budget)
+  print(paste("remaining budget: ", remaining_budget, " seconds"))
+  if (type == "classification") {
+    train <- TaskClassif$new("benchmark_train", backend = train, target = target)
+    test <- TaskClassif$new("benchmark_test", backend = test, target = target)
+    if ("twoclass" %in% train$properties) {
+      measure = msr("classif.auc")
+    } else {
+      measure = msr("classif.logloss")
+    }
+    model <- AutoML(train, runtime = as.integer(remaining_budget * 0.8), measure = measure)
+  } else if (type == "regression") {
+    train <- TaskRegr$new("benchmark_train", backend = train, target = target)
+    test <- TaskRegr$new("benchmark_test", backend = test, target = target)
+    model <- AutoML(train, runtime = as.integer(remaining_budget * 0.8))
+  } else {
+    stop("Task type not supported!")
+  }
+  print(paste("Finished creating model after ", difftime(Sys.time(), start_time, units = "secs"), " seconds"))
+  model$train()
+  print(paste("Finished training model after ", difftime(Sys.time(), start_time, units = "secs"), " seconds"))
+  preds <- model$predict(test)
```

Comment:
training duration and prediction duration should now be captured.
See `ranger` or `autoxgboost` for examples of how this can be done in R.